decoder.go

```diff
@@ -114,7 +119,7 @@ func newAVFrameWrapper() (*avFrameWrapper, error) {
 	if avFrame == nil {
 		return nil, errors.New("failed to allocate AVFrame: out of memory or C libav internal error")
 	}
-	wrapper := &avFrameWrapper{frame: avFrame}
+	wrapper := &avFrameWrapper{frame: avFrame, generation: -1}
```

Comment:
You could consider having `newAVFrameWrapper` taking a generation as input. But I'm not going to block approval on this.